### PR TITLE
impl binary op for array (not ref)

### PR DESCRIPTION
**Pull request by minghuaw**
### Dependencies

- [x] #76

### CODE REVIEW

**Comment by minghuaw** [comment]
This PR has been updated to #76

File: examples/tutorial.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ fn array_basics() {
     let y = Array::ones::<f32>(&[2, 2]).unwrap();
 
     // Pointwise add x and y:
-    let z = x.add(&y);
+    let z = x.as_ref().add(&y);
```
Comment (dcvz) [inline examples/tutorial.rs:43]: is this change related?

**Review by dcvz — APPROVED** [review]
LGTM 👍

File: examples/tutorial.rs (hunk 1 above)
Comment (minghuaw) [reply examples/tutorial.rs]: I thought the compiler would complain because `x` is not a ref, but `x.add(&y)` actually works fine. This change is reverted in [ac32a53](https://github.com/oxideai/mlx-rs/pull/71/commits/ac32a53e652fc819ef8a885f51e4a28c6231ba55)
